libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/alert/RouteAlert.kt

```diff
@@ -110,56 +105,163 @@ class TunnelEntranceAlert private constructor(
     }
 
     /**
-     * Use to create a new instance.
-     *
-     * @see TunnelEntranceAlert
+     * Metadata specific to this alert. Currently contains no additional data.
      */
-    class Builder(
-        private val metadata: Metadata,
```

Comment:
the reason you have semver changes with this builder, is because the Builder has required parameters.

Make all of them optional, with default values. That makes it so you can add new variables with default values without breaking semver

Reply:
Resolving this conversation in favor of this thread. Considering these domain objects are generated by the SDK unlike domain objects like NavigationOptions (which come from external clients)

There's good reason to remove builders. And consider better ways to create these objects internally.

Moved thread to over here https://github.com/mapbox/mapbox-navigation-android/pull/3582#discussion_r492357006